src/rgw/rgw_pubsub.h

```diff
@@ -675,81 +642,12 @@ class RGWPubSub
     int remove_notifications(const DoutPrefixProvider *dpp, optional_yield y);
   };
 
-  // base class for subscription
-  class Sub {
-    friend class RGWPubSub;
-  protected:
-    RGWPubSub* const ps;
-    const std::string sub;
-    rgw_raw_obj sub_meta_obj;
-
-    int read_sub(rgw_pubsub_sub_config *result, RGWObjVersionTracker* objv_tracker);
-    int write_sub(const DoutPrefixProvider *dpp, const rgw_pubsub_sub_config& sub_conf,
-		  RGWObjVersionTracker* objv_tracker, optional_yield y);
-    int remove_sub(const DoutPrefixProvider *dpp, RGWObjVersionTracker* objv_tracker, optional_yield y);
-  public:
-    Sub(RGWPubSub *_ps, const std::string& _sub) : ps(_ps), sub(_sub) {
-      ps->get_sub_meta_obj(sub, &sub_meta_obj);
-    }
-
-    virtual ~Sub() = default;
-
-    int subscribe(const DoutPrefixProvider *dpp, const std::string& topic_name, const rgw_pubsub_sub_dest& dest, optional_yield y,
-		  const std::string& s3_id="");
-    int unsubscribe(const DoutPrefixProvider *dpp, const std::string& topic_name, optional_yield y);
-    int get_conf(rgw_pubsub_sub_config* result);
-    
-    static const int DEFAULT_MAX_EVENTS = 100;
-    // followint virtual methods should only be called in derived
-    virtual int list_events(const DoutPrefixProvider *dpp, const std::string& marker, int max_events) {ceph_assert(false);}
-    virtual int remove_event(const DoutPrefixProvider *dpp, const std::string& event_id) {ceph_assert(false);}
-    virtual void dump(Formatter* f) const {ceph_assert(false);}
-  };
-
-  // subscription with templated list of events to support both S3 compliant and Ceph specific events
-  template<typename EventType>
-  class SubWithEvents : public Sub {
-  private:
-    struct list_events_result {
-      std::string next_marker;
-      bool is_truncated{false};
-      void dump(Formatter *f) const;
-      std::vector<EventType> events;
-    } list;
-
-  public:
-    SubWithEvents(RGWPubSub *_ps, const std::string& _sub) : Sub(_ps, _sub) {}
-
-    virtual ~SubWithEvents() = default;
-    
-    int list_events(const DoutPrefixProvider *dpp, const std::string& marker, int max_events) override;
-    int remove_event(const DoutPrefixProvider *dpp, const std::string& event_id) override;
-    void dump(Formatter* f) const override;
-  };
-
   using BucketRef = std::shared_ptr<Bucket>;
```

Comment:
might want to consider removing the 'using' from the h file.

Reply:
i don't understand what is the issue with this definition here?
BTW, it would probably be good idea to remove this shared pointer altogether (i don't see why it is useful)